python/note_input_widget/widget.py

```diff
@@ -46,14 +49,18 @@ class NoteInputWidget(QtGui.QWidget):
     data_updated = QtCore.Signal()
     close_clicked = QtCore.Signal()
 
+    # Emitted when a new note or reply will be created. The creation request id
+    # is sent and will match the id sent with the entity_created signal.
+    # Todo: This has functional overlap with the pre_submit_callback...resolve that...
+    new_entity_requested = QtCore.Signal(str, int)
```

Comment:
Your approach with new signals seems justified and appropriate to me Ed. Looks good